src/Passes/CONTRIBUTING.md

```diff
@@ -0,0 +1,83 @@
+# Contributing (Proposal - WiP)
+
+This document is work in progress and nothing is set in stone. In case you do not want to feel like reading this style guide, just run
+
+```sh
+./manage runci
+```
+
+from the `Passes` directory as all points defined in this document is automatically enforces. You can then refer to this guide for an explanation for why and how.
+
+## Why do we need a style guide?
+
+Consistency and readibility such that it is easy to read and understand code that was not written by yourself. For example, if one developer uses `CamelCase` for namespaces and `snake_case` for classes while another uses `snake_case` for namespaces and `CamelCase` you may end up with code sections that looks like this
+
+```cpp
+int32_t main()
+{
+  name_space1::Class1 hello;
+  NameSpace2::class_name world;
+}
+```
+
+which is hard to read.
+
+## What does the style guide apply to?
+
+The style guide applies to any new code written as well as code that is being refactored added to the `Passes` library. We do not rewrite existing code for the sake just changing the style.
+
+## Style discrepency
+
+In case of a discrepency between this guideline and `clang-tidy` or `clang-format`,
+clang tools rule. In case of discrency between this guide and any guides subsequently referenced guides, this guide rule. However, feel free to suggest changes. Changes will be incorporated on the basis
+that updated styles are apply to new code and not existing code.
+
+## Naming
+
+Naming is taken from the [Microsoft AirSim](https://github.com/microsoft/AirSim/blob/master/docs/coding_guidelines.md) project.
+
+| **Code Element**      | **Style**                        | **Comment**                                                                                                                                   |
+| --------------------- | -------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| Namespace             | snake_case                       | Differentiates `namespace::ClassName` and `ClassName::SubClass` names                                                                         |
```

Comment:
Yes, I believe that we use CamelCase in other places. However, that practice lowers the readability of the code since you cannot differentiate namespaces from classes by looking at the name. Instead, you need to look up the definitions as they explain in the `AirSim` style guide. As this is seemed like a sensible argument and given that there are minor differences between the current style and that which is used in other places, I suggested this change when we defined the style.

With that said, if people strongly feel that namespaces should be `CamelCase`, I am happy to make the change.